internal/util/fetch/fetch.go

```diff
@@ -94,21 +94,62 @@ func (c Command) validate(kf *kptfilev1.KptFile) error {
 	return nil
 }
 
+type Cloner struct {
+	// repoSpec spec to clone
+	repoSpec *git.RepoSpec
+
+	// cachedRepos
+	cachedRepo map[git.RepoSpec]*gitutil.GitUpstreamRepo
+
+	// fetchedRefs keeps track of refs already fetched from remote
+	fetchedRefs map[string]bool
```

Comment:
cachedRepo above also has `fetchedRefs` stored ...right ?

Reply:
Right, I have removed this.